src/Features/Core/Portable/Completion/MatchResult.cs

```diff
@@ -5,53 +5,53 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using Microsoft.CodeAnalysis.Completion;
 using Microsoft.CodeAnalysis.PatternMatching;
-using Microsoft.VisualStudio.Text;
 using Roslyn.Utilities;
 using RoslynCompletionItem = Microsoft.CodeAnalysis.Completion.CompletionItem;
-using VSCompletionItem = Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion.Data.CompletionItem;
 
-namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncCompletion
+namespace Microsoft.CodeAnalysis.Completion
 {
-    internal readonly struct MatchResult
+    internal readonly struct MatchResult<T>
     {
         public readonly RoslynCompletionItem RoslynCompletionItem;
         public readonly bool MatchedFilterText;
-        public readonly ImmutableArray<Span> HighlightedSpans;
```

Comment:
still unclear how this can be moved off the MatchResult itself.

Reply:
The pattern match result contains the matched spans for the roslyn item.  These highlighted spans are the VS spans that correspond to the VS item's display text.  We can just delay the calculation of roslyn matched span -> VS completion item span to when the VS FilteredCompletionModel is created from the MatchResult

Reply:
gotcha.  what is the value in doing that though?